x11-wm/cwm updated to latest 6.6 and new proxy-maintainer

epsilon-0 opened the pull request:
Package has been stale and using out of date version from 7 years ago.

Adding myself as proxy-maintainer

Bug: https://bugs.gentoo.org/718682

Signed-off-by: Aisha Tammy <gentoo@aisha.cc>

gyakovlev commented on x11-wm/cwm/cwm-6.6.ebuild:
```diff
@@ -1,35 +1,34 @@
+	KEYWORDS=""
+else
+	SRC_URI="https://github.com/chneukirchen/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+	KEYWORDS="amd64 arm64 x86"
```
you can't add new ebuild with stable keywords

epsilon-0 replied:
Yep, I've changed it back to the unstable ones, and I've also readded the `~arm` keyword I removed by mistake.

gyakovlev commented on x11-wm/cwm/cwm-6.6.ebuild:
```diff
@@ -1,35 +1,34 @@
-	fi
-}
+	sys-devel/bison
+"
```
those dependencies are usually BDEPEND in EAPI=7

epsilon-0 replied:
Totally forgot, thanks for noticing, I've fixed that :)

gyakovlev commented on x11-wm/cwm/cwm-6.6.ebuild:
```diff
@@ -1,35 +1,34 @@
-# Copyright 1999-2018 Gentoo Authors
+# Copyright 2020 Gentoo Authors
```
you should leave original year intact.

epsilon-0 replied:
I've updated the years to the correct range, hopefully it is OK now.

gyakovlev commented on x11-wm/cwm/cwm-6.6.ebuild:
```diff
@@ -1,35 +1,34 @@
+EAPI=7
 
-inherit eutils toolchain-funcs
+inherit desktop eutils toolchain-funcs
```
I don't see eutils being used in the ebuild

epsilon-0 replied:
I've removed it from the ebuild, thanks again

gyakovlev commented on x11-wm/cwm/cwm-6.6.ebuild:
```diff
@@ -1,35 +1,34 @@
+
+if [[ ${PV} == 9999 ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/chneukirchen/${PN}.git"
```
this url is incorrect (as is down below)
It's https://github.com/leahneukirchen/cwm

epsilon-0 replied:
I've updated the link, thanks a lot :)

jonasstein commented:
Please change
Bug: https://bugs.gentoo.org/718682
to
Closes: https://bugs.gentoo.org/718682

epsilon-0 commented:
> We don't usually remove old ebuilds on big bumps. Leave old ebuild in place and introduce new one and live as new files.
> 

I've added the old one back :)

> Also makefile calls pkg-config directly, it'll break cross-compilation.
> Will be nicer if calls to pkg-config will be replaced by proper call.
> for example, modify Makefile to add
> 
> `PKG_CONFIG?= pkg-coinfig`
> 
> and have ebuild define it as `PKG_CONFIG="$(tc-getPKG_CONFIG)"` , similar to how `CC` is defined on command line and in Makefile.

I've made a pull request upstream so that this will be easier next release onwards.
Until then, I've added a patch, hopefully that should be OK :)

Thanks a lot for all your comments @gyakovlev

epsilon-0 commented:
> Please change
> Bug: https://bugs.gentoo.org/718682
> to
> Closes: https://bugs.gentoo.org/718682

I've changed that to the correct one, thanks a lot for the information @jonasstein :)

gyakovlev commented on x11-wm/cwm/cwm-6.6.ebuild:
```diff
@@ -0,0 +1,45 @@
+"
+BDEPEND="${RDEPEND}
+	virtual/pkgconfig
+	sys-devel/bison
```
small nit, sort alphabetically. I can do it on merge.

epsilon-0 replied:
Haha, I've sorted them. Should look cleaner now :)

gyakovlev commented on x11-wm/cwm/cwm-6.6.ebuild:
```diff
@@ -0,0 +1,45 @@
+	x11-libs/libXinerama
+	x11-libs/libXrandr
+"
+BDEPEND="${RDEPEND}
```
no, this is incorrect.
you should define
DEPEND="${RDEPEND}",
BDEPEND should only contain bison and pkgconfig

gyakovlev replied:
here's some documentation on differences between DEPEND and BDEPEND
https://dev.gentoo.org/~mgorny/articles/the-ultimate-guide-to-eapi-7.html#build-time-dependencies-split-into-depend-and-bdepend

epsilon-0 replied:
Thanks a lot for the link!

I've split them correctly.

gyakovlev commented:
I merged it with some minor fixes, closing manually. thanks!

gyakovlev commented:
https://github.com/gentoo/gentoo/commit/e84ae9943a70dcac2434c18533327df73b471a71
